geographical-analysis.csl

```diff
@@ -0,0 +1,178 @@
+  <info>
+    <title>Geographical Analysis</title>
+    <title-short>GEAN</title-short>
+    <id>http://www.zotero.org/styles/geographical-analysis</id>
+    <link href="http://www.zotero.org/styles/geographical-analysis" rel="self"/>
+    <link href="http://www.zotero.org/styles/environment-and-planning" rel="template"/>
+    <link href="https://onlinelibrary.wiley.com/page/journal/15384632/homepage/forauthors.html" rel="documentation"/>
+    <author>
+      <name>Yigong Hu</name>
+    </author>
+    <category citation-format="author-date"/>
+    <category field="geography"/>
+    <issn>0016-7363</issn>
+    <summary>Style for the Geographical Analysis journal</summary>
+    <updated>2015-01-10T18:11:11+00:00</updated>
+    <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>
+  </info>
+  <locale xml:lang="en">
+    <style-options punctuation-in-quote="true"/>
+    <terms>
+      <term name="editor" form="verb-short">edited by</term>
+      <term name="editor" form="short">
+        <single>ed</single>
+        <multiple>eds</multiple>
+      </term>
+      <term name="in">In</term>
+      <term name="edition">ed.</term>
+    </terms>
+  </locale>
+  <macro name="editor">
+    <names variable="editor" delimiter="," suffix=".">
+      <label form="verb-short" suffix=" "/>
+      <name sort-separator=", " and="text" initialize-with="." delimiter=", "/>
+    </names>
+  </macro>
+  <macro name="author">
+    <names variable="author">
+      <name name-as-sort-order="all" sort-separator=" " initialize-with=". " delimiter=", " and="text"/>
+      <label form="short" prefix=" "/>
+      <substitute>
+        <names variable="editor"/>
+        <text variable="title"/>
+      </substitute>
+    </names>
+  </macro>
+  <macro name="author-short">
+    <names variable="author">
+      <name form="short" and="text" delimiter=", " initialize-with="."/>
+      <substitute>
+        <names variable="editor"/>
+        <names variable="translator"/>
+        <text variable="title"/>
+      </substitute>
+    </names>
+  </macro>
+  <macro name="title">
+    <choose>
+      <if type="book thesis" match="any">
+        <text variable="title" font-style="italic" suffix=". "/>
+      </if>
+      <else>
+        <text variable="title" quotes="true" suffix=". "/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher">
+    <choose>
+      <if type="book chapter" match="any">
+        <group delimiter=": " prefix="" suffix="">
+          <text variable="publisher-place"/>
+          <text variable="publisher"/>
+        </group>
+      </if>
+      <else-if type="paper-conference" match="any">
+        <group delimiter=": " prefix="">
+          <text variable="publisher-place"/>
+          <text variable="publisher"/>
+        </group>
+      </else-if>
+      <else>
+        <group delimiter=", " prefix=", ">
```

Comment:
This prefix is going to produce some weird results with the prefix that is included when the publisher macro is called in the bibliography. I suggest removing the prefixes here and then converting them to groups and delimiters in the bibliography. 

Please do this throughout the style when you are using prefixes or suffixes as delimiters between elements. 

Also, do you mean for journal, newspaper, and magazine articles to have their publishers included? If not, please specify what types you mean, rather than using `else`?